File: schemas/recipe-cooking/drizzle/collections.ts
// collections: User-curated groups of recipes with visibility and cover images.
// See README.md for full design rationale.

import { pgTable, uuid, text, integer, boolean, timestamp, index } from "drizzle-orm/pg-core";
import { users } from "../../auth-rbac/drizzle/users";

export const collections = pgTable(
  "collections",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    name: text("name").notNull(),
    description: text("description"),
    coverImageUrl: text("cover_image_url"),
    isPublic: boolean("is_public").notNull().default(false),
    recipeCount: integer("recipe_count").notNull().default(0),
    createdBy: uuid("created_by").notNull().references(() => users.id, { onDelete: "cascade" }),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
    updatedAt: timestamp("updated_at", { withTimezone: true }).notNull().defaultNow(),
  },
  (table) => [
    index("idx_collections_created_by").on(table.createdBy),
  ]
);

File: schemas/recipe-cooking/spacetimedb/collections.rs
// collections: User-curated groups of recipes.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = collections, public)]
pub struct Collection {
    #[primary_key]
    pub id: String, // UUID

    pub name: String,
    pub description: Option<String>,
    pub cover_image_url: Option<String>,
    pub is_public: bool, // default false
    pub recipe_count: i32, // default 0

    #[index(btree)]
    pub created_by: String, // FK → users.id

    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}
